centraldogma/src/main/java/com/linecorp/decaton/centraldogma/CentralDogmaPropertySupplier.java

```diff
@@ -145,19 +148,43 @@ public static CentralDogmaPropertySupplier register(CentralDogma centralDogma, S
         return new CentralDogmaPropertySupplier(centralDogma, project, repository, filename);
     }
 
+    public static CentralDogmaPropertySupplier register(CentralDogma centralDogma, String project,
+                                                        String repository, String filename,
+                                                        PropertySupplier supplier) {
+        List<Property<?>> properties = defaultProperties().stream().map(defaultProperty -> {
+            if (supplier.getProperty(defaultProperty.definition()).isPresent()) {
+                return supplier.getProperty(defaultProperty.definition()).get();
+            } else {
+                return defaultProperty;
+            }
+        }).collect(Collectors.toList());
+
+        createPropertyFile(centralDogma, project, repository, filename, properties);
+        return new CentralDogmaPropertySupplier(centralDogma, project, repository, filename);
+    }
+
     private static void createPropertyFile(CentralDogma centralDogma, String project,
```

Comment:
I think this overload no longer meaningful. Let's just use the below signature and pass default properties directly.

Reply:
I deleted this original method `createPropertyFile` (no longer used), since a new method which has more arguments was added. https://github.com/line/decaton/pull/134/commits/6b49b94c9a1f6cb7493fb3f13be8d4240f4c1dff